### ISSUE TITLE
Removing target groups from aws_autoscaling_group does not remove it on AWS

### ISSUE THREAD

**Issue by JoelSpeed** [terraform-providers/terraform-provider-aws#3265]
I've tested this a couple of times. Attaching an NLB to the ASG using a target group and then removing it again, terraform does not remove it from AWS

### Terraform Version
v0.11.1

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_autoscaling_group

### Terraform Configuration Files
```hcl
resource "aws_autoscaling_group" "controllers" {
  name                      = "${var.cluster_name}-controllers"
  launch_configuration      = "${aws_launch_configuration.controllers.name}"
  min_size                  = "${var.min_controllers}"
  max_size                  = "${var.max_controllers}"
  health_check_type         = "EC2"
  health_check_grace_period = 600
  wait_for_capacity_timeout = "10m"
  load_balancers            = ["${aws_elb.apiserver_elb.name}"]
  target_group_arns         = ["${aws_lb_target_group.apiserver_lb_internal.arn}"]
  vpc_zone_identifier       = ["${aws_subnet.cluster_subnets.*.id}"]

  enabled_metrics = ["GroupDesiredCapacity", "GroupInServiceInstances", "GroupStandbyInstances", "GroupTotalInstances"]

  tag {
    key                 = "KubernetesCluster"
    value               = "${var.cluster_name}"
    propagate_at_launch = true
  }

  tag {
    key                 = "Role"
    value               = "Controller"
    propagate_at_launch = true
  }
}
```

### Expected Behavior
Terraform should have removed the target group from the ASG on AWS

### Actual Behavior
The ASG remained attached to the ASG

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
2. comment out `target_group_arns`
3. `terraform apply` - nothing to change

**Comment by MichaelMure** [comment]
Can confirm the problem. The same way, `lifecycle.ignore_changes` seems to **not** be replicated either.

> Terraform v0.11.11
> provider.aws v1.54.0

**Comment by anGie44** [comment]
The associated changes have been merged and will release with the upcoming `v3.0.0` release of the Terraform AWS Provider.
